docs/TRACEZ_DESIGN.md

```diff
@@ -0,0 +1,175 @@
+# OpenTelemetry SDK Contrib - /tracez and /traceconfigz Design Doc
+
+This file contains information about the design choices for the OpenTelemetry /tracez and
+/traceconfigz zPages.
+
+## Introduction
+
+The OpenTelemetry zPages are a set of dynamically generated HTML pages that display trace and
+metrics data for a running process. Within this set, the /tracez zPage is a web page that displays
+trace information about running spans, sample span latencies, and sample error spans. On the other
+hand, the /traceconfigz zPage is a web page that allows the users to change tracing parameters, such
+as sampling probability and max number of attributes.
+
+### /tracez zPage
+
+The /tracez zPage displays information on running spans, sample span latencies, and sample error
+spans. The data is aggregated into a summary-level table:
+
+![tracez-table](img/tracez-table.png)
+
+In addition, you can click on each of the counts in the table cells to access the corresponding span
+details. For example, here are the details of the `ChildSpan` latency sample (row 1, col 4):
+
+![tracez-details](img/tracez-details.png)
+
+### /traceconfigz zPage
+
+The /traceconfigz zPage displays information about the currently active tracing configuration and 
+provides an interface for users to modify relevant parameters:
+
+![traceconfigz](img/traceconfigz.png)
+
+## Motivation
+
+At the moment, there are no zPages for the Java implementation of OpenTelemetry. The goal of this
```

Comment:
I think we can shrink this to a single sentence: We want to create a tool that lets users troubleshoot OpenTelemetry instrumentation that can also be used as a lightweight application performance monitoring tool.

The other stuff seems like it could become outdated. "In addition, the implementation of the Java zPages can pave the way for future implementations in other languages, such as Python, which have yet to be built."